tools/python/transit/transit_graph_generator.py

```diff
@@ -0,0 +1,365 @@
+                guide1 = self.__get_average_stops_point(info['guide_points'][id1])
+            else:
+                guide1 = [2 * point1[0] - point2[0], 2 * point1[1] - point2[1]]
+
+            if info['guide_points'][id2]:
+                guide2 = self.__get_average_stops_point(info['guide_points'][id2])
+            else:
+                guide2 = [2 * point2[0] - point1[0], 2 * point2[1] - point1[1]]
+
+            curve_points = bezier_curves.segment_to_Catmull_Rom_curve(guide1, point1, point2, guide2,
+                                                                      self.points_per_curve, self.alpha)
+            info['curve'] = np.array(curve_points)
+
+            polyline = []
+            for point in curve_points:
+                polyline.append({'x': point[0], 'y': point[1]})
+
+            shape = {'id': shape_id,
+                     'stop1_id': id1,
+                     'stop2_id': id2,
+                     'polyline': polyline}
+            self.shapes.append(shape)
+
+        for edge in self.edges:
+            if not edge['transfer']:
+                stop1 = self.stops[edge['start_stop_id']]
+                stop2 = self.stops[edge['finish_stop_id']]
+                id1 = stop1.get('transfer_id', stop1['id'])
+                id2 = stop2.get('transfer_id', stop2['id'])
+                seg = tuple(sorted([id1, id2]))
+                if seg in self.segments:
+                    edge['shape_ids'].append(self.segments[seg]['shape_id'])
+
+    def build(self):
+        if self.transit_graph is not None:
+            return self.transit_graph
+
+        self.__read_stops()
+        self.__read_transfers()
+        self.__read_networks()
+        self.__generate_transfer_nodes()
+        self.__generate_shapes()
+
+        self.transit_graph = {'networks': self.networks,
+                              'lines': self.lines,
+                              'gates': self.gates.values(),
+                              'stops': self.stops.values(),
+                              'transfers': self.transfers.values(),
+                              'shapes': self.shapes,
+                              'edges': self.edges}
+        return self.transit_graph
+
+    def show_preview(self):
+        for (s1, s2), info in self.segments.items():
+            plt.plot(info['curve'][:, 0], info['curve'][:, 1], 'g')
+        for stop in self.stops.values():
+            if 'transfer_id' in stop:
+                point = self.transfers[stop['transfer_id']]['point']
+                size = 60
+                color = 'r'
+            else:
+                point = stop['point']
+                if len(stop['line_ids']) > 2:
+                    size = 40
+                    color = 'b'
+                else:
+                    size = 20
+                    color = 'g'
+            plt.scatter([point['x']], [point['y']], size, color)
+        plt.show()
+
+
+def check_file_exists(file_path):
+    if not os.path.isfile(file_path):
```

Comment:
Why not just to use exceptions and maybe even have no check of this kind at all.